test_scripts/Polices/user_consent_of_Policies/ATF_Device_identifier_creation_upon_connect_with_apps.lua

```diff
@@ -42,64 +44,72 @@ function Test:Precondition_Connect_device()
-  {
-    application =
-    {
-      deviceInfo =
-      {
-        name = "127.0.0.1",
-        id = config.deviceMAC,
-        transportType = "WIFI"
-      }
-    }
-  })
-  :Do(function(_,data)
-  self.HMIAppID = data.params.application.appID
-  end)
-  self.mobileSession:ExpectResponse(RequestIDRai1, { success = true, resultCode = "SUCCESS" })
-  self.mobileSession:ExpectNotification("OnHMIStatus", {hmiLevel = "NONE", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN"})
-  end)
+      local RequestIDRai1 = self.mobileSession:SendRPC("RegisterAppInterface", config.application1.registerAppInterfaceParams)
+      EXPECT_HMINOTIFICATION("BasicCommunication.OnAppRegistered",
+        {
+          application =
+          {
+            deviceInfo =
+            {
+              name = "127.0.0.1",
+              id = config.deviceMAC,
+              transportType = "WIFI"
+            }
+          }
+        })
+      :Do(function(_,data)
+          self.applications[config.application1.registerAppInterfaceParams.appName] = data.params.application.appID
+        end)
+      self.mobileSession:ExpectResponse(RequestIDRai1, { success = true, resultCode = "SUCCESS" })
+      self.mobileSession:ExpectNotification("OnHMIStatus", {hmiLevel = "NONE", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN"})
+    end)
+end
+
+function Test:Precondition_TriggerGettingDeviceConsent()
+  testCasesForPolicyTable:trigger_getting_device_consent(self, config.application1.registerAppInterfaceParams.appName, config.deviceMAC)
 end
 
 --[[ Test ]]
 commonFunctions:newTestCasesGroup("Test")
 function Test:Check_device_identifier_added_to_lpt()
   local is_test_fail = true
-  local file = io.open(pts_json, "r")
-  local json_data = file:read("*all") -- may be abbreviated to "*a";
-  file:close()
-  local json = require("modules/json")
-  local data = json.decode(json_data)
-  local deviceIdentificatorInPTS = next(data.policy_table.device_data, nil)
-  if (deviceIdentificatorInPTS == config.deviceMAC) then
-    commonFunctions:userPrint(33, "device_identifier ".. deviceIdentificatorInPTS.. " section is created")
-    is_test_fail = false
-  end
-  if(is_test_fail == true) then
-    self:FailTestCase("Test is FAILED.")
+  if( commonSteps:file_exists(pts_json) == false) then
+    self:FailTestCase(pts_json .. " doesn't exist")
+  else
+    local file = io.open(pts_json, "r")
+    local json_data = file:read("*all") -- may be abbreviated to "*a";
+    file:close()
+    local json = require("modules/json")
+    local data = json.decode(json_data)
+    local deviceIdentificatorInPTS = next(data.policy_table.device_data, nil)
+    if (deviceIdentificatorInPTS == config.deviceMAC) then
+      commonFunctions:userPrint(33, "device_identifier ".. deviceIdentificatorInPTS.. " section is created")
+      is_test_fail = false
+    end
+    if(is_test_fail == true) then
+      self:FailTestCase("Test is FAILED.")
+    end
```

Comment:
@istoimenova, Test step can be failed directly, without of usage of _is_test_fail_ varibale